payments/src/middlewares/auth.middleware.ts

```diff
@@ -24,8 +26,26 @@ const authMiddleware = async (req: RequestWithUser, res: Response, next: NextFun
       next(new TokenError(404, 'Authentication token missing'));
     }
   } catch (error) {
-    next(new TokenError(401, 'Wrong authentication token'));
+    next(error);
   }
 };
 
+const extractAuthorizationHeader = (req: RequestWithUser) => {
+  const header = req.header('Authorization');
+  return header ? header.split('Bearer ')[1] : null;
```

Comment:
What happens if malicious user sends in the header: `Bearer ` 
You will have exception here